Dockerfile

```diff
@@ -0,0 +1,17 @@
+# Use an official Python runtime as a parent image
```

Comment:
Can we try a multistage build to reduce the image size, as we are running it on a low-spec machine?

```
# Use an official Python runtime as a parent image and Build environment
FROM python:3.8-slim AS build

# Set the working directory to /app
WORKDIR /app

# Copy the current directory contents into the container at /app
COPY . /app

# Install any needed packages specified in requirements.txt
RUN pip install --no-cache-dir -r requirements.txt

# Stage 2: Production environment
FROM python:3.8-slim AS production

# Set the working directory to /app
WORKDIR /app

# Copy only the necessary files from the build stage to the production image
COPY --from=build /app /app

# Expose port 8000 for the Gunicorn server to listen on
EXPOSE 8000

# Define the command to run your application using Gunicorn
CMD ["gunicorn", "server:app", "--bind", "0.0.0.0:8000", "--workers", "4"]

```

Reply:
I will check if multistage will make much difference here, mostly multistage is vital when you need to install other "apt" packages to install the packages. I should add the `--no-cache-dir` though.

Reply:
Switched to alpine, using a multi-stage build is helpful when you install other packages to install the main dependencies.